lib/segment/src/index/field_index/full_text_index/inverted_index.rs

```diff
@@ -132,20 +134,33 @@ impl InvertedIndex {
         query: &ParsedQuery,
         condition: &FieldCondition,
     ) -> CardinalityEstimation {
-        let (points_count, index_postings) = match self {
-            InvertedIndex::Mutable(index) => (index.points_count, &index.postings),
-            InvertedIndex::Immutable(index) => (index.points_count, &index.postings),
+        let points_count = match self {
+            InvertedIndex::Mutable(index) => index.points_count,
+            InvertedIndex::Immutable(index) => index.points_count,
         };
-        let postings_opt: Option<Vec<_>> = query
+        let posting_lengths: Option<Vec<usize>> = query
             .tokens
             .iter()
             .map(|&vocab_idx| match vocab_idx {
                 None => None,
                 // unwrap safety: same as in filter()
-                Some(idx) => index_postings.get(idx as usize).unwrap().as_ref(),
+                Some(idx) => match &self {
+                    Self::Mutable(index) => index
+                        .postings
+                        .get(idx as usize)
+                        .unwrap()
+                        .as_ref()
+                        .map(|p| p.len()),
+                    Self::Immutable(index) => index
+                        .postings
+                        .get(idx as usize)
+                        .unwrap()
+                        .as_ref()
+                        .map(|p| p.len()),
+                },
             })
             .collect();
```

Comment:
I probably need an extra explanation why we have double Option around `PostingList`?

And how Option of `p.len()` is collected into just `Vec<usize>`? Do we skip Nones? If we have no posting lists for token, shouldn't we consider it as length=0?

Reply:
Agree that this `Option`s are unclear. I think that we can simplify the whole inverted index and remove a lot of obsolete `Option`s. But right now I'm not sure that there are no any place where `Option` is important. I Propose to fix it as separate PR, created issue for that:
https://github.com/qdrant/qdrant/issues/3716